serviceerror/convert.go

```diff
@@ -71,7 +72,21 @@ func FromStatus(st *status.Status) error {
 		return nil
 	}
 
-	// Simple case. `st.Code()` to `serviceerror` is one to one mapping and there are no error details.
+	errDetails := extractErrorDetails(st)
```

Comment:
The error parsing had to be changed so that it always parses the error details first. This adds an extra step for status codes such as Unavailable and Canceled, but is most likely negligible.